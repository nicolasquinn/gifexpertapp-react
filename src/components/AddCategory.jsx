import { useState } from "react"
import { PropTypes } from "prop-types";

export const AddCategory = ({ onAddCategory }) => {

  const [ inputValue, setInputValue ] = useState('');

  const onInputChange = ({ target }) => {

    setInputValue( target.value );

  }

  const onFormSubmit = (e) => {

    e.preventDefault();

    if ( inputValue.trim().length <= 1 ) return;

    onAddCategory( inputValue.trim() );
    setInputValue('');
    
  }

  return (
    <form onSubmit={ onFormSubmit } aria-label="form">
      <input 
        type="text"
        placeholder="Search GIFs"
        value={ inputValue }
        onChange={ onInputChange }
      />
    </form>
  )
}

AddCategory.propTypes = {
  onAddCategory: PropTypes.func.isRequired
}
