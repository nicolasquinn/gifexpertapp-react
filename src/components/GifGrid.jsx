import { getGifs } from "../helpers/getGifs";

// Se recibe de manera implÃ­cita el key.
export const GifGrid = ({ category }) => {

  getGifs(category);

  return (
    <>
        <h3>{ category }</h3>
        <p>La imagen</p>
    </>
  )
}
