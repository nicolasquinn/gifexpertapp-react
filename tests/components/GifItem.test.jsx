import { getByRole, render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Testing GifItem component', () => {

    const title = 'Just a title';
    const url = 'https://arandomurl.com/'

    test('Snapshot should match', () => {

        const { container } = render( <GifItem title={ title } url={ url } /> );
        expect( container ).toMatchSnapshot();

    });

    test('Must show the image with the correct URL and ALT text', () => {

        render( <GifItem title={ title } url={ url } /> );

        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );

        const { alt, src } = screen.getByRole('img');
        expect( alt ).toBe( title );
        expect( src ).toBe( url );

    });

    test('Property "title" must be into the component as a text', () => {

        render( <GifItem title={ title } url={ url } /> );
        expect( screen.getByText(title) ).toBeTruthy();

    });

})