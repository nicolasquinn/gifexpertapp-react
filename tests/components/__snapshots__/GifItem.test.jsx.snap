// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifItem component Snapshot should match 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Just a title"
      src="https://arandomurl.com/"
    />
    <p>
      Just a title
    </p>
  </div>
</div>
`;
